http-client-javanet/src/main/java/io/micronaut/http/client/javanet/AbstractJavanetHttpClient.java

```diff
@@ -0,0 +1,232 @@
+            .cookieHandler(cookieManager);
+
+        Optional<SocketAddress> proxyAddress = configuration.getProxyAddress();
+        if (proxyAddress.isPresent()) {
+            SocketAddress socketAddress = proxyAddress.get();
+            builder = configureProxy(builder, socketAddress, configuration.getProxyUsername().orElse(null), configuration.getProxyPassword().orElse(null));
+        }
+
+        if (configuration.getSslConfiguration() instanceof ClientSslConfiguration clientSslConfiguration) {
+            configureSsl(builder, clientSslConfiguration);
+        }
+
+        this.client = builder.build();
+    }
+
+    private HttpClient.Builder configureProxy(
+        @NonNull HttpClient.Builder builder,
+        @NonNull SocketAddress address,
+        @Nullable String username,
+        @Nullable String password
+    ) {
+        if (log.isDebugEnabled()) {
+            log.debug("Configuring proxy: {} with username: {}", address, username);
+        }
+        if (address instanceof InetSocketAddress inetSocketAddress) {
+            builder = builder.proxy(ProxySelector.of(inetSocketAddress));
+            if (username != null && password != null) {
+                builder = builder.authenticator(new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(username, password.toCharArray());
+                    }
+                });
+            }
+        } else {
+            throw new IllegalArgumentException("Unsupported proxy address type: " + address.getClass().getName());
+        }
+        return builder;
+    }
+
+    private void configureSsl(HttpClient.Builder builder, ClientSslConfiguration clientSslConfiguration) {
+        sslBuilder.build(clientSslConfiguration).ifPresent(builder::sslContext);
+        if (System.getProperty("jdk.internal.httpclient.disableHostnameVerification") != null && log.isWarnEnabled()) {
+            log.warn("The jdk.internal.httpclient.disableHostnameVerification system property is set. This is not recommended for production use as it prevents proper certificate validation and may allow man-in-the-middle attacks.");
+        }
+        SSLParameters sslParameters = new SSLParameters();
+        clientSslConfiguration.getClientAuthentication().ifPresent(a -> {
+            if (a == ClientAuthentication.WANT) {
+                sslParameters.setWantClientAuth(true);
+            } else if (a == ClientAuthentication.NEED) {
+                sslParameters.setNeedClientAuth(true);
+            }
+        });
+        clientSslConfiguration.getProtocols().ifPresent(sslParameters::setProtocols);
+        clientSslConfiguration.getCiphers().ifPresent(sslParameters::setCipherSuites);
+        builder.sslParameters(sslParameters);
+    }
+
+    protected Object getLoadBalancerDiscriminator() {
+        return null;
+    }
```

Comment:
Since this is a new implementation, just remove this and pass the request as the discriminator. See https://github.com/micronaut-projects/micronaut-core/issues/8823